stout/flags.h

```diff
@@ -36,24 +38,40 @@ class Parser {
   template <typename Flags>
   friend class ParserBuilder;
 
+  // For building nested parsers.
+  template <typename Flags>
+  static ParserBuilder<Flags> Builder(
+      Flags* flags,
+      bool is_nested) {
+    return ParserBuilder<Flags>(flags, Parser{is_nested});
+  }
+
   Parser()
     : standard_flags_(new stout::v1::StandardFlags()) {
     // Add all the "standard flags" first, e.g., --help.
-    AddAllOrExit(standard_flags_.get());
+    TryFillFieldsAndMessages(standard_flags_.get());
   }
 
-  // Helper that adds all the flags and their descriptors.
-  void AddAllOrExit(google::protobuf::Message* message);
+  // Constructor for nested parsers.
+  Parser(bool is_nested)
+    : standard_flags_(new stout::v1::StandardFlags()),
+      is_nested_(is_nested) {
+    // Add all the "standard flags" for nested messages too,
+    // e.g., --help.
+    TryFillFieldsAndMessages(standard_flags_.get());
+  }
+
+  // Helper that fill fields_ and messages_ helpers.
+  void TryFillFieldsAndMessages(google::protobuf::Message* message);
 
-  // Helper that adds a flag and it's descriptor.
-  void AddOrExit(
-      const std::string& name,
-      const google::protobuf::FieldDescriptor* field,
-      google::protobuf::Message* message);
+  // Helper that gets 'google::protobuf::Message*' based on
+  // subcommand argument from the command line.
+  google::protobuf::Message*
+  GetMessageForSubcommand(const std::string& arg);
 
   // Helper for parsing a normalized form of flags.
-  void Parse(
-      const std::multimap<std::string, std::optional<std::string>>& values);
+  void
+  Parse(const std::multimap<std::string, std::optional<std::string>>& values);
```

Comment:
Wrap at the argument if necessary, otherwise:
```suggestion
  void Parse(const std::multimap<std::string, std::optional<std::string>>& values);
```